main_test.go

```diff
@@ -3,152 +3,209 @@ package main
+	if actual != expected {
+		t.Errorf("TestParseContent_DirectText: Expected %s but got %s", expected, actual)
 	}
 }
 
-func TestCopyFromCommandLineArgument(t *testing.T) {
-	if testing.Short() {
-		t.Skip("skipping clipboard test in CI/CD.")
-	}
+// TestParseContent_Stdin verifies parseContent handles stdin input correctly.
+func TestParseContent_Stdin(t *testing.T) {
+	// Stdin input content
+	content := "Stdin input\n"
+	expected := "Stdin input\n"
 
-	content := "Command line text"
-	os.Args = []string{"clipper", "-c", content}
+	// Replace stdin with a pipe for testing
+	_, w, cleanup := replaceStdin(t)
+	defer cleanup()
 
-	main()
+	// Simulate writing content to stdin
+	go func() {
+		w.Write([]byte(content))
+		w.Close()
+	}()
 
-	copied, err := clipboard.ReadAll()
+	// Execute the function under test with stdin input
+	actual, err := parseContent(nil, []string{})
 	if err != nil {
-		t.Errorf("Failed to read from clipboard: %v", err)
+		t.Fatalf("TestParseContent_Stdin: Expected no error, got %v", err)
 	}
 
-	if copied != content {
-		t.Errorf("Expected %s but got %s", content, copied)
+	// Verify the actual output matches the expected output
+	if actual != expected {
+		t.Errorf("TestParseContent_Stdin: Expected %s but got %s", expected, actual)
 	}
 }
 
-func TestParseContentDirectText(t *testing.T) {
-	directText := "Direct text input"
-	contentStr, err := parseContent(&directText, []string{})
+// TestParseContent_File verifies parseContent handles file input correctly.
+func TestParseContent_File(t *testing.T) {
+	// Content to write to the temporary file
+	content := "Content from file"
+	file := createTempFile(t, content)
+	defer os.Remove(file.Name()) // Clean up: remove the temporary file after the test
+
+	// Expected content after parsing the file
+	expected := "Content from file\n"
+
+	// Execute the function under test with the temporary file as input
+	actual, err := parseContent(nil, []string{file.Name()})
 	if err != nil {
-		t.Errorf("Expected no error, got %v", err)
+		t.Fatalf("TestParseContent_File: Expected no error, got %v", err)
 	}
-	if contentStr != directText {
-		t.Errorf("Expected contentStr to be '%s', got '%s'", directText, contentStr)
+
+	// Verify the actual output matches the expected output
+	if actual != expected {
+		t.Errorf("TestParseContent_File: Expected %s but got %s", expected, actual)
 	}
 }
 
-func TestParseContentStdin(t *testing.T) {
-	stdinInput := "Stdin input\n"
-	r, w, _ := os.Pipe()
-	originalStdin := os.Stdin
-	defer func() {
-		os.Stdin = originalStdin
-		r.Close()
-		w.Close()
-	}()
-	os.Stdin = r
-
-	// Write the input to the pipe
-	w.WriteString(stdinInput)
-	w.Close()
+// stringPtr creates a pointer to a string.
+func stringPtr(s string) *string {
+	return &s
+}
 
-	contentStr, err := parseContent(nil, []string{})
+// createTempFile creates a temporary file for testing purposes.
+func createTempFile(t *testing.T, content string) *os.File {
```

Comment:
Missing t.Helper call here


```suggestion
func createTempFile(t *testing.T, content string) *os.File {
t.Helper ()
```